Frameworks/CoreImage/CIImage.mm

```diff
@@ -73,90 +74,164 @@ @implementation CIImage
 /**
  @Status Interoperable
 */
-- (CIImage*)init {
++ (CIImage*)emptyImage {
+    return [[[CIImage alloc] init] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (CGRect)extent {
+    return self->_extent;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)init {
     if (self = [super init]) {
-        _cgImage = nil;
-        _color = nil;
-        _filter = nil;
         _extent = CGRectInfinite;
     }
     return self;
 }
 
+- (instancetype)_initWithCGImage:(CGImageRef)image color:(CIColor*)color {
+    if (self = [self initWithCGImage:image]) {
+        _color.attach([static_cast<id>(color) copy]);
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithColor:(CIColor*)color {
+    if (!color) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [self init]) {
+        _color.attach([static_cast<id>(color) copy]);
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)cgImage {
+    if (!cgImage) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [self init]) {
+        _cgImage.reset(CGImageRetain(cgImage));
+        _extent = CGRectMake(0, 0, CGImageGetWidth(cgImage), CGImageGetHeight(cgImage));
+    }
+
+    return self;
+}
+
 /**
  @Status Stub
  @Notes
 */
-+ (CIImage*)emptyImage {
+- (instancetype)initWithCGImage:(CGImageRef)image options:(NSDictionary*)d {
     UNIMPLEMENTED();
     return StubReturn();
 }
 
 /**
  @Status Interoperable
 */
-+ (CIImage*)imageWithColor:(CIColor*)color {
-    CIImage* ret = [[CIImage alloc] init];
-    if (ret != nil) {
-        ret->_color.attach([static_cast<id>(color) copy]);
-        ret->_extent = CGRectMake(INFINITY, INFINITY, CGFLOAT_MAX, CGFLOAT_MAX);
+- (instancetype)initWithImage:(UIImage*)image {
+    if (!image) {
```

Comment:
i'll leave it as is for now. it's more explicit rather than a chain.
both ways work fine

---
In reply to: [95705497](https://github.com/Microsoft/WinObjC/pull/1625#discussion_r95705497) [](ancestors = 95705497)